pallets/connectors/src/lib.rs

```diff
@@ -378,38 +442,36 @@ pub mod pallet {
 		#[pallet::call_index(5)]
 		pub fn update_member(
 			origin: OriginFor<T>,
-			domain_address: DomainAddress,
 			pool_id: PoolIdOf<T>,
 			tranche_id: TrancheIdOf<T>,
+			domain_address: DomainAddress,
 			valid_until: Moment,
 		) -> DispatchResult {
 			let who = ensure_signed(origin.clone())?;
 
-			// Check that the origin is allowed to add other members
 			ensure!(
-				T::Permission::has(
-					PermissionScope::Pool(pool_id),
-					who.clone(),
-					Role::PoolRole(PoolRole::InvestorAdmin)
-				),
-				BadOrigin
+				T::PoolInspect::pool_exists(pool_id),
+				Error::<T>::PoolNotFound
+			);
+			ensure!(
+				T::PoolInspect::tranche_exists(pool_id, tranche_id),
+				Error::<T>::TrancheNotFound
+			);
+			ensure!(
+				valid_until > Self::now(),
+				Error::<T>::InvalidTrancheInvestorValidity
 			);
 
-			// Now add the destination address as a TrancheInvestor of the given tranche if
-			// not already one. This check is necessary shall a user have called
-			// `update_member` already but the call has failed on the EVM side and needs to
-			// be retried.
-			if !T::Permission::has(
-				PermissionScope::Pool(pool_id),
-				T::AccountConverter::convert(domain_address.clone()),
-				Role::PoolRole(PoolRole::TrancheInvestor(tranche_id, valid_until)),
-			) {
-				T::Permission::add(
+			// Ensure that the destination address has been whitelisted as a TrancheInvestor
+			// beforehand.
+			ensure!(
+				T::Permission::has(
 					PermissionScope::Pool(pool_id),
 					T::AccountConverter::convert(domain_address.clone()),
-					Role::PoolRole(PoolRole::TrancheInvestor(tranche_id, valid_until)),
-				)?;
-			}
+					Role::PoolRole(PoolRole::TrancheInvestor(tranche_id, valid_until))
```

Comment:
Did we alredy adapt the behaviour of the permissions checking to take into account the `valid_until`? ^^

Reply:
Lazy me, did not read carefully enough. Sorry ^^